docs/components/docgen.js

```diff
@@ -43,16 +43,3 @@ export default async function docgen(component: string): Promise<DocGen> {
 
   return parsed;
 }
-
-export function overrideTypes(docGen: DocGen, typeOverrides: {| [string]: string |}): DocGen {
```

Comment:
Given that this functionality is used in several places, is there a reason to remove this helper function and inline the logic?

Reply:
@dangerismycat I added it back in to unblock this PR. Originally I inlined it for the following reasons:

* The function has the wrong abstraction: it always assumes that the type is `union` (for module I needed `string`). We could add another prop to allow for a different type but that only makes the function more complex.
* `typeOverrides` felt like a weird format, I was expecting `{ name: 'Icon', raw: '', name: '' }`

In general I feel like we're making our code base a bit too DRY by over-abstracting code which makes it harder to see what the code is supposed to do.

More resources on this topic:
* https://www.deconstructconf.com/2019/dan-abramov-the-wet-codebase
* https://www.philosophicalhacker.com/post/when-to-dry/

Reply:
Totally valid reason — I think you could have left these changes in the PR. We can re-add them in a follow-up

Reply:
Also: I get the over-abstraction pain as well. I was just glancing through the Button types the other day and was so confused. It felt like that tweet where Abramov was revisiting Redux and thought it was way too complex 😅